pure-pulsars/src/cmds/wikianimal.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import os
+import re
+
+import aiohttp
+import discord
+from discord import NotFound, app_commands
+from discord.app_commands.errors import CommandInvokeError
+from pint import UnitRegistry
+
+import button_class
+from button_class import GameType, GiveUpButton, GuessButton, _Button
+from wikiutils import UA, make_img_embed, search_wikipedia
+
+UREG = UnitRegistry()
+
+WEIGHT_PATTERN = re.compile(r"(\d+ ?(?:kg|tons|lbs|oz|g|pounds|grams))")
+
+
+class IncompatibleAnimalError(Exception):
+    """Error to be thrown if wikipedia animal and/or animal api fail to provide a valid weight range."""
+
+    def __init__(self) -> None:
+        super().__init__("No valid animal weights")
+
+
+class WinLossFunctions(button_class.WinLossManagement):
+    """The Basic Win Loss Function for WikiAnimal."""
+
+    def __init__(self, winargs: dict, lossargs: dict) -> None:
+        super().__init__(winargs, lossargs)
+
+    async def on_win(self) -> None:
+        """Clean up on win."""
+
+    async def on_loss(self) -> None:
+        """Clean up on loss."""
+
+
+async def find_animal_weight(animal_name: str) -> list[str]:
+    """Determine animal's weight ranges based on article text."""
+    api_url = f"https://api.api-ninjas.com/v1/animals?name={animal_name}"
+    async with (
+        aiohttp.ClientSession(timeout=aiohttp.ClientTimeout(total=20)) as session,
+        session.get(api_url, headers={"X-Api-Key": os.environ["NINJA_API_KEY"]}) as response,
+    ):
+        if not response.ok:
+            logging.error("Error requesting animal data: %s", response.status)
+            raise IncompatibleAnimalError
+        animal_data = await response.json()
+    if animal_data:
+        weight_str = animal_data[0].get("characteristics").get("weight")
+    else:
+        raise IncompatibleAnimalError
+    if weight_str is None:
+        raise IncompatibleAnimalError
+    weight_str = weight_str.replace(",", "")
+    weight_phrases = re.findall(WEIGHT_PATTERN, weight_str)
+    if weight_phrases:
+        weight_ranges = [UREG.Quantity(weight) for weight in weight_phrases]
+        if len(weight_ranges) % 2 == 0:
+            return [weight_ranges[0], weight_ranges[1]]
```

Comment:
I don't understand the point of this code, why are we checking if its even/odd? Could we adda  comment